File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }


        int lockTimerCounter;
        private void OnTimerTick(object sender, System.EventArgs e)
        {
            if (lockTimerCounter == 0)
            {
                lockTimerCounter++;
                UpdateText();
                lockTimerCounter--;
            }
        }
        protected void Init()
        {
            timer.Stop();
            offsetCounter = -digitalGauge1.DigitCount;
            timer.Interval = 500 / (trackBarControl1.Value + 1);
            timer.Start();
        }
        int offsetCounter = 0;
        protected void UpdateText()
        {
            string fullTextToShow = textEdit1.Text;
            char[] textToShow = new char[digitalGauge1.DigitCount];
            for (int i = 0; i < digitalGauge1.DigitCount; i++)
            {
                if (i + offsetCounter >= 0 && i + offsetCounter < fullTextToShow.Length)
                {
                    textToShow[i] = fullTextToShow[i + offsetCounter];
                }
                else textToShow[i] = ' ';
            }
           offsetCounter++; 
            if (offsetCounter > digitalGauge1.DigitCount + fullTextToShow.Length) offsetCounter = -digitalGauge1.DigitCount;
            digitalGauge1.Text = new string(textToShow);
        }


        private void textEdit1_EditValueChanged(object sender, EventArgs e)
        {
            Init();
        }


        private void trackBarControl1_ValueChanged(object sender, EventArgs e)
        {
            Init();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            new XtraForm1().ShowProgress();
        }

    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub


		Private lockTimerCounter As Integer
		Private Sub OnTimerTick(ByVal sender As Object, ByVal e As System.EventArgs) Handles timer.Tick
			If lockTimerCounter = 0 Then
				lockTimerCounter += 1
				UpdateText()
				lockTimerCounter -= 1
			End If
		End Sub
		Protected Sub Init()
			timer.Stop()
			offsetCounter = -digitalGauge1.DigitCount
			timer.Interval = 500 / (trackBarControl1.Value + 1)
			timer.Start()
		End Sub
		Private offsetCounter As Integer = 0
		Protected Sub UpdateText()
			Dim fullTextToShow As String = textEdit1.Text
			Dim textToShow(digitalGauge1.DigitCount - 1) As Char
			For i As Integer = 0 To digitalGauge1.DigitCount - 1
				If i + offsetCounter >= 0 AndAlso i + offsetCounter < fullTextToShow.Length Then
					textToShow(i) = fullTextToShow.Chars(i + offsetCounter)
				Else
					textToShow(i) = " "c
				End If
			Next i
		   offsetCounter += 1
			If offsetCounter > digitalGauge1.DigitCount + fullTextToShow.Length Then
				offsetCounter = -digitalGauge1.DigitCount
			End If
			digitalGauge1.Text = New String(textToShow)
		End Sub


		Private Sub textEdit1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles textEdit1.EditValueChanged
			Init()
		End Sub


		Private Sub trackBarControl1_ValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackBarControl1.EditValueChanged
			Init()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			CType(New XtraForm1(), XtraForm1).ShowProgress()
		End Sub

	End Class
End Namespace